dissect/volume/md/stream.py

```diff
@@ -0,0 +1,394 @@
+from __future__ import annotations
+
+import io
+import operator
+from typing import TYPE_CHECKING, BinaryIO, NamedTuple, Optional
+
+from dissect.util.stream import AlignedStream, MappingStream
+
+from dissect.volume.exceptions import MDError
+from dissect.volume.md.c_md import c_md
+
+if TYPE_CHECKING:
+    from dissect.volume.md.md import MD, Device
+
+
+def create_linear_stream(md: MD) -> BinaryIO:
```

Comment:
Would it be possible to have this as a `MappingStream` subclass? That would make the API for all the types of MD arryas more consistent.